guillotina/db/transaction.py

```diff
@@ -43,9 +43,10 @@ class Status:
 
 
 class cache:
-    def __init__(self, key_gen, check_state_size=False):
+    def __init__(self, key_gen, check_state_size=False, additional_keys=None):
```

Comment:
Apparently `key_gen` and `additional_keys` have the same purpose, I think we will have a more consistent API if we just use `key_gen` as a 
typing.Union[typing.Callable, typing.List[typing.Callable]].

I think we should add a docstring to describe the signature call to expect for the decorated function. That will really help new contributors like myself.

Reply:
This is a little bit of a difference in that `key_gen` is used to get it from the parameters, `additional_keys` sets the key/value based on the value itself.

So they are different signatures.

But yes, type hints will help and maybe I can improve the naming as well.